dandiapi/api/management/commands/import_dandisets.py

```diff
@@ -0,0 +1,137 @@
+            )
+        ).json()
+
+        click.echo(f'  Importing version "{result["version"]}"')
+
+        version = Version(
+            dandiset=dandiset,
+            name=result['name'],
+            version=result['version'],
+            doi=result.get('doi'),
+            status=result['status'],
+            metadata=metadata,
+        )
+        version.save()
+
+        # validate the metadata after the transaction is commmited
+        transaction.on_commit(lambda: validate_version_metadata.delay(version.id))
+
+    # Handle API pagination
+    if version_api_response.get('next'):
+        next_versions = requests.get(version_api_response.get('next')).json()
+        import_versions_from_response(api_url, next_versions, dandiset)
+
+
+@transaction.atomic
+def import_dandiset_from_response(
+    api_url: str, dandiset_api_response: dict, dandiset_to_replace=None
+):
+    identifier = dandiset_api_response['identifier']
+
+    # If replacing a dandiset, find the existing one and delete it first
+    if dandiset_to_replace is not None:
+        Dandiset.objects.get(id=dandiset_to_replace).delete()
+        dandiset = Dandiset(id=dandiset_to_replace)
+    else:
+        dandiset = Dandiset()
+
+    dandiset.save()
+
+    # get all versions associated with this dandiset
+    versions = requests.get(urljoin(api_url, f'/api/dandisets/{identifier}/versions/')).json()
+
+    click.echo(
+        f'Importing dandiset {identifier} "{dandiset_api_response["draft_version"]["name"]}" as dandiset {dandiset.identifier}'  # noqa: E501
+    )
+
+    import_versions_from_response(api_url, versions, dandiset)
+
+
+@transaction.atomic
+def import_dandisets_from_response(api_url: str, dandiset_api_response: dict):
+    """Import dandisets given a response from /api/dandisets/."""
+    for result in dandiset_api_response['results']:
+        import_dandiset_from_response(api_url, result)
+
+    # Handle API pagination
+    if dandiset_api_response.get('next'):
+        new_dandisets = requests.get(dandiset_api_response.get('next')).json()
+        import_dandisets_from_response(
+            api_url,
+            new_dandisets,
+        )
+
+
+@click.command()
+@click.argument('api_url')
+@click.option('--all', is_flag=True, required=False, help='Download all dandisets.')
+@click.option('--identifier', required=False, help='The identifier of the dandiset to import.')
+@click.option('--replace', default=None, help='The dandiset to replace with the imported one.')
+@click.option('--offset', default=0, help="Offset to add to each new dandiset's identifier.")
+@transaction.atomic
+def import_dandisets(api_url: str, all: bool, identifier: str, replace: str, offset=0):
+    """Click command to import dandisets from another server deployment."""
+    # Save the current postgres sequence value and set it with the offset
+    original_sequence_value = get_sequence_value()
+    set_sequence_value(original_sequence_value + offset)
```

Comment:
This is not something I thought we would be including in the script. I thought that `offset` would default to `200000`, and would be added individually to each dandiset identifier as it was imported. If you only import dandiset `000036` from production, this script would save that new dandiset at `200036`. 

Regardless, if I used this script to download a single dandiset metadata to reproduce something locally or whatever, I would certainly not expect it to reset my dandiset table sequence.

Reply:
> This is not something I thought we would be including in the script. I thought that `offset` would default to `200000`, and would be added individually to each dandiset identifier as it was imported. If you only import dandiset `000036` from production, this script would save that new dandiset at `200036`.

I guess I was thinking about it differently, but your way seems more logical and way easier. I'll go ahead and implement it that way and remove those SQL queries

> Doesn't the script reset it to its original value at the end? Though your approach avoids the problem of the script crashing before it can do so, leaving the DB in the wrong state.

FWIW, yes it does reset it to the original value at the end. And there's no possibility that the DB would end up in a wrong state since the entire script is wrapped in a transaction (see `transaction.atomic` decorator).

Reply:
> > Doesn't the script reset it to its original value at the end? Though your approach avoids the problem of the script crashing before it can do so, leaving the DB in the wrong state.
> 
> FWIW, yes it does reset it to the original value at the end. And there's no possibility that the DB would end up in a wrong state since the entire script is wrapped in a transaction (see `transaction.atomic` decorator).

Ah nice touch, I had missed that!